Source/WebCore/accessibility/cocoa/AXTextMarkerCocoa.mm

```diff
@@ -119,9 +119,10 @@
     auto* cache = object->axObjectCache();
     if (!cache)
         return nil;
-    return adoptNS(@[
+    return adoptNS([[NSArray alloc] initWithObjects:
         adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_start.m_data cache:cache]).autorelease(),
-        adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_end.m_data cache:cache]).autorelease()
+        adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_end.m_data cache:cache]).autorelease(),
+        nil
     ]);
```

Comment:
This will fix the bug, but you could also avoid autoreleasing the individual items in the array by doing this instead:

```diff
    auto start = adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_start.m_data cache:cache]);
    auto end = adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_end.m_data cache:cache]);
    return adoptNS([[NSArray alloc] initWithObjects:start.get(), end.get(), nil]);
```

Here the `start` and `end` objects will have `-release` called when the function returns, but not before building the `NSArray` with them, which itself will take a +1 reference to each object.

Generally speaking, it's best to avoid autoreleasing an Objective-C object if it can be avoided as it prevents work that must be done later to clear the autoreleasePool.

Reply:
Thanks very much, took the suggestion as well.